### DISCUSSION TITLE
SMILE interpreter in python

### DISCUSSION THREAD

**Discussion by BiocharModeling** [deepmodeling/reacnetgenerator#2087] [Q&A] [answered]
I hope you are doing well. I was wondering if you could clarify to me how the program read the SMILES. Based on the results, for example, the SMILE of the group OH ([**H]O**) is interpreted in Python as H2O when I post-process the data. But in the HTML file, this SMILE is drawn as OH. What interpretation are you using for the SMILE in Python?
 
Thank you!

**Comment by njzjz** [comment, marked as answer]
The SMILES in Python is generated by RDKit. Sadly, per https://github.com/rdkit/rdkit/discussions/6613, RDKit fills all atoms with implicit Hs, and there's no option to disable it.

The SMILES in HTML is converted by a regex rule

https://github.com/tongzhugroup/reacnetgenerator/blob/b5a16eb90cdf1eca03738218eba5044d2f151fec/reacnetgenerator/_reachtml.py#L69-L99

**Reply by njzjz** [reply]
Perhaps the conversion process should happen before Python results are outputed.

**Reply by BiocharModeling** [reply]
Thank you!!

**Reply by njzjz** [reply]
> Perhaps the conversion process should happen before Python results are outputted.

Fixed in #2102
